include/swift/Demangling/Demangle.h

```diff
@@ -216,8 +216,21 @@ class Node {
   void dump();
 };
 
+/// Returns the length of the swift mangling prefix of the \p SymbolName.
+///
+/// Returns 0 if \p SymbolName is not a mangled swift (>= swift 4.x) name.
+int getManglingPrefixLength(const char *mangledName);
```

Comment:
It could, but I don't see the benefit of it.
I don't see the need of Optional<StringRef> because the None case could simply be an empty StringRef. But then, why returning the (char_ptr, size) (= StringRef) and not just the size if the char-ptr is not used? We would introduce the implicit assumption that the returned StringRef is always the prefix of the argument. Well, I prefer to just return the size.